ui-tests/src/test/java/io/syndesis/qe/pages/app/AppPage.java

```diff
@@ -0,0 +1,117 @@
+package io.syndesis.qe.pages.app;
+
+import static com.codeborne.selenide.Selenide.$;
+
+import org.apache.log4j.Logger;
+
+import org.openqa.selenium.By;
+
+import com.codeborne.selenide.CollectionCondition;
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+
+import java.util.Map;
+
+import io.syndesis.qe.pages.interfaces.SyndesisComponent;
+
+/**
+ * Created by sveres on 11/8/17.
+ */
+public class AppPage implements SyndesisComponent {
+
+	private static final Logger log = Logger.getLogger(AppPage.class);
+
+	private static final class Element {
+		public static final SelenideElement ROOT = $(By.cssSelector("syndesis-root"));
+	}
+
+	@Override
+	public SelenideElement rootElement() {
+		return Element.ROOT.shouldBe(Condition.exist);
+	}
+
+	private static String baseurl = "/";
+
+	public SelenideElement getButton(String buttonTitle) {
+		log.info(String.format("searching for button %s", buttonTitle));
+		return this.rootElement().find(By.xpath(String.format("//button[contains(text(), '%s')]", buttonTitle)));
+	}
+
+	//IS THIS METHOD NECESSARY? - YES IT IS!! KEEP IT.
+	public void clickOnFirstVisibleButton(String buttonTitle) {
+
+		log.info(String.format("searching for first visible button %s", buttonTitle));
+		ElementsCollection buttonElements = this.rootElement().findAll(By.linkText(buttonTitle)).filter(Condition.visible);
```

Comment:
It would be nice to have all these locator objects statically defined similar way as root locator.